### PR TITLE
Migrate CollectionUtilsTest to Kotlin

### PR DESCRIPTION
**Pull request by oyeraghib**
## Purpose / Description
Migrate CollectionUtilsTest to Kotlin

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.kt [hunk 1]
```diff
@@ -13,41 +13,33 @@
...
-        add(3);
-    }};
+class CollectionUtilsTest {
+    var testList = arrayListOf(1, 2, 3)
 
     @Test
-    public void testGetLastListElement() {
-        assertThat(CollectionUtils.getLastListElement(mTestList), is(3));
+    fun testGetLastListElement() {
+        MatcherAssert.assertThat(getLastListElement(testList), CoreMatchers.`is`(3))
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.kt:30]: kotlincleanup is -> equalto

**Review by mikehardy — CHANGES_REQUESTED** [review]
Thanks!
one kotlincleanup notation needed I think

File: AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.kt [hunk 2]
```diff
@@ -13,41 +13,34 @@
...
-    }};
+class CollectionUtilsTest {
+    var testList = arrayListOf(1, 2, 3)
 
     @Test
-    public void testGetLastListElement() {
-        assertThat(CollectionUtils.getLastListElement(mTestList), is(3));
+    @KotlinCleanup("is --> equalTo")
+    fun testGetLastListElement() {
+        MatcherAssert.assertThat(getLastListElement(testList), CoreMatchers.`is`(3))
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.kt:31]: import `MatcherAssert`, `CoreMatchers`

File: AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.kt [hunk 3]
```diff
@@ -13,41 +13,34 @@
...
 
     @Test
-    public void testAddAll() {
-        List<Integer> toTest = new ArrayList<>();
-        CollectionUtils.addAll(toTest, mTestList);
-        assertEqualsArrayList(new Integer [] {1, 2, 3}, toTest);
+    fun testAddAll() {
+        val toTest = arrayListOf<Int>()
+        addAll(toTest, testList)
+        AnkiAssert.assertEqualsArrayList(arrayOf(1, 2, 3), toTest)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.kt:44]: import: `AnkiAssert`

File: AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.kt [hunk 4]
```diff
@@ -13,41 +13,34 @@
  *  You should have received a copy of the GNU General Public License along with
  *  this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+package com.ichi2.utils
 
-package com.ichi2.utils;
+import com.ichi2.testutils.AnkiAssert
+import com.ichi2.utils.CollectionUtils.addAll
+import com.ichi2.utils.CollectionUtils.getLastListElement
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.kt:20]: Remove these two lines and use `CollectionUtils.method` at the call site (to match the Java)

**Review by david-allison — CHANGES_REQUESTED** [review]

**Review by david-allison — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
excellent

**Pull request merged**